src/libraries/System.Text.Json/tests/Common/RequiredKeywordTests.cs

```diff
@@ -610,6 +610,23 @@ public class ClassWithRequiredExtensionDataProperty
             public required Dictionary<string, JsonElement>? TestExtensionData { get; set; }
         }
 
+        [Fact]
+        public async Task RequiredCollectionPropertyWithoutGetterThrows()
+        {
+            string json = """{"Foo":"foo","Bar":"bar"}""";
+            InvalidOperationException exception = await Assert.ThrowsAsync<InvalidOperationException>(
+                async () => await Serializer.DeserializeWrapper<ClassWithRequiredCollectionPropertyWithoutGetter>(json));
+            Assert.Contains(nameof(ClassWithRequiredCollectionPropertyWithoutGetter.TestCollectionPropertyWithoutGetter), exception.Message);
+        }
+
+        public class ClassWithRequiredCollectionPropertyWithoutGetter
```

Comment:
Just to make sure we're not regressing things here, could you also add a test for class like so:
```C#
class ClassWithCollectionCtorParam(Dictionary<string, JsonElement> value)
{
    public Dictionary<string, JsonElement> Value { get; } = value;
}
```
And then try to deserialize it using an `options` instance with `RespectRequiredConstructorParameters` set to `true`.